nifi-nar-bundles/nifi-datadog-bundle/nifi-datadog-reporting-task/src/main/java/org/apache/nifi/reporting/datadog/DataDogReportingTask.java

```diff
@@ -0,0 +1,185 @@
+            .name("DataDog reporting period")
+            .description("The amount of time in seconds between polls")
+            .required(true)
+            .expressionLanguageSupported(false)
+            .defaultValue("10")
+            .addValidator(StandardValidators.LONG_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor METRICS_PREFIX = new PropertyDescriptor.Builder()
+            .name("Metrics prefix")
+            .description("Prefix to be added before every metric")
+            .required(true)
+            .expressionLanguageSupported(false)
+            .defaultValue("nifi")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor ENVIRONMENT = new PropertyDescriptor.Builder()
+            .name("Environment")
+            .description("Environment, dataflow is running in. " +
+                    "This property will be included as metrics tag.")
+            .required(true)
+            .expressionLanguageSupported(false)
+            .defaultValue("dev")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    private MetricsService metricsService;
+    private DDMetricRegistryBuilder ddMetricRegistryBuilder;
+    private MetricRegistry metricRegistry;
+    private String metricsPrefix;
+    private String environment;
+    private String statusId;
+    private ConcurrentHashMap<String, AtomicDouble> metricsMap;
+    private volatile VirtualMachineMetrics virtualMachineMetrics;
+    private Logger logger = LoggerFactory.getLogger(getClass().getName());
+
+    @OnScheduled
+    public void setup(final ConfigurationContext context) throws IOException {
+        metricsService = getMetricsService();
+        ddMetricRegistryBuilder = getMetricRegistryBuilder();
+        metricRegistry = getMetricRegistry();
+        metricsMap = getMetricsMap();
+        metricsPrefix = METRICS_PREFIX.getDefaultValue();
+        environment = ENVIRONMENT.getDefaultValue();
+        virtualMachineMetrics = VirtualMachineMetrics.getInstance();
+        ddMetricRegistryBuilder.setMetricRegistry(metricRegistry)
+                .setName("nifi_metrics")
+                .setTags(Arrays.asList("env", "dataflow_id"))
+                .build();
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(REPORTING_PERIOD);
+        properties.add(METRICS_PREFIX);
+        properties.add(ENVIRONMENT);
+        return properties;
+    }
+
+    @Override
+    public void onTrigger(ReportingContext context) {
+        final ProcessGroupStatus status = context.getEventAccess().getControllerStatus();
+        final String reportingPeriod = context.getProperty(REPORTING_PERIOD)
+                .evaluateAttributeExpressions().getValue();
```

Comment:
You are running "evaluateAttributeExpressions" on a property that doesn't support expression language.